### PR TITLE
Nope viewer

### PR DESCRIPTION
**Pull request by ubitux**

### CODE REVIEW

File: pynopegl-utils/pynopegl_utils/qml/ngl_widget.py [hunk 1]
```diff
@@ -52,7 +52,11 @@ def createRenderer(self):
         return self._renderer
 
     def set_scene(self, scene):
-        model_data = livectls.get_model_data(scene)
+        if scene:
```
Comment (mbouron) [inline pynopegl-utils/pynopegl_utils/qml/ngl_widget.py:55]: Why is this useful while we have the stop method ?

Comment (ubitux) [reply pynopegl-utils/pynopegl_utils/qml/ngl_widget.py]: Mmh yeah I thought I still had something like that but that commit is not needed anymore so dropped

File: pynopegl-utils/pynopegl_utils/qml/viewer.qml [hunk 2]
```diff
@@ -0,0 +1,503 @@
...
+        id: scriptDialog
+        objectName: "scriptDialog"
+        nameFilters: ["Python scripts (*.py)"]
+        onAccepted: {
+            script.text = selectedFile
+            script.editingFinished()
+        }
+    }
+
+
```
Comment (mbouron) [inline pynopegl-utils/pynopegl_utils/qml/viewer.qml:48]: Extra line.

File: pynopegl-utils/pynopegl_utils/viewer/config.py [hunk 3]
```diff
@@ -0,0 +1,179 @@
...
+    name: str
+    format: str
+    args: List[str]
+
+
+ENCODE_PROFILES = dict(
+    mp4_h264_420=EncodeProfile(
+        name="MP4 / H264 4:2:0",
+        format="mp4",
+        args=["-pix_fmt", "yuv420p"],
```
Comment (mbouron) [inline pynopegl-utils/pynopegl_utils/viewer/config.py:54]: I would explicitly specify the libx264 encoder even though it is the default for mp4 (if libx264 is enabled). I would also specify a good enough (tm) crf.

File: pynopegl-utils/pynopegl_utils/qml/viewer.qml [hunk 4]
```diff
@@ -0,0 +1,503 @@
...
+    }
+
+    function start_export(filename, res, profile, samples) {
+        exportBar.value = 0
+        exportStack.currentIndex = 0
+        exportPopup.visible = true
+        exportVideo(filename, res, profile, samples)
+    }
+
+    function done_export() {
```
Comment (mbouron) [inline pynopegl-utils/pynopegl_utils/qml/viewer.qml:125]: `done_export()` -> `finish_export()` ? (so we have start/finish)

File: pynopegl-utils/pynopegl_utils/qml/viewer.qml [hunk 5]
```diff
@@ -0,0 +1,503 @@
...
+        exportStack.currentIndex = 1
+    }
+
+    function error_export(msg) {
```
Comment (mbouron) [inline pynopegl-utils/pynopegl_utils/qml/viewer.qml:129]: `abort_export(msg)` to use a verb and be consistent with the other

File: pynopegl-utils/pynopegl_utils/viewer/__init__.py [hunk 6]
```diff
@@ -0,0 +1,543 @@
...
+    @Slot(int)
+    def _select_framerate(self, index: int):
+        framerate = self._config.CHOICES["framerate"][index]
+        self._config.set_framerate(framerate)
+        self._load_current_scene()
+
+    @Slot()
+    def _set_export_filename(self):
+        filename = self._export_filename_text.property("text")
+        filename = QUrl(filename).path()  # handle the file:// automatically added
```
Comment (mbouron) [inline pynopegl-utils/pynopegl_utils/viewer/__init__.py:203]: + by Qt

File: pynopegl-utils/pynopegl_utils/qml/viewer.qml (hunk 2 above)
Comment (ubitux) [reply pynopegl-utils/pynopegl_utils/qml/viewer.qml]: Removed

File: pynopegl-utils/pynopegl_utils/viewer/config.py (hunk 3 above)
Comment (ubitux) [reply pynopegl-utils/pynopegl_utils/viewer/config.py]: 1. Added `-c:v libx264`
2. What value for CRF? 21? 18?
3. faststart added (420 only), with a comment on the rationale

File: pynopegl-utils/pynopegl_utils/qml/viewer.qml (hunk 4 above)
Comment (ubitux) [reply pynopegl-utils/pynopegl_utils/qml/viewer.qml]: Renamed

File: pynopegl-utils/pynopegl_utils/qml/viewer.qml (hunk 5 above)
Comment (ubitux) [reply pynopegl-utils/pynopegl_utils/qml/viewer.qml]: Renamed

File: pynopegl-utils/pynopegl_utils/viewer/__init__.py (hunk 6 above)
Comment (ubitux) [reply pynopegl-utils/pynopegl_utils/viewer/__init__.py]: Added

**Review by mbouron — APPROVED** [review]

**Pull request merged**